src/main/java/com/conveyal/datatools/editor/controllers/api/RouteController.java

```diff
@@ -28,82 +32,70 @@
 
 
 public class RouteController {
-    public static JsonManager<Route> json =
+    public static final JsonManager<Route> json =
             new JsonManager<>(Route.class, JsonViews.UserInterface.class);
     private static final Logger LOG = LoggerFactory.getLogger(RouteController.class);
+
     public static Object getRoute(Request req, Response res) {
         String id = req.params("id");
         String feedId = req.queryParams("feedId");
-        Object json = null;
-
-        if (feedId == null)
-            feedId = req.session().attribute("feedId");
 
         if (feedId == null) {
             halt(400);
         }
 
-        final FeedTx tx = VersionedDataStore.getFeedTx(feedId);
+        FeedTx tx = null;
 
         try {
+            tx = VersionedDataStore.getFeedTx(feedId);
             if (id != null) {
                 if (!tx.routes.containsKey(id)) {
-                    tx.rollback();
                     halt(400);
                 }
 
                 Route route = tx.routes.get(id);
                 route.addDerivedInfo(tx);
 
-                json = Base.toJson(route, false);
-
-//                return route;
+                return route;
             }
             else {
-                Route[] ret = tx.routes.values().toArray(new Route[tx.routes.size()]);
+                Set<Route> ret = new HashSet<>(tx.routes.values());
```

Comment:
The set works fine, but what was wrong with what we had before? toArray will create an array divorced from the underlying database.